From a886bddc2a9d4e2859b653c136d43c1f9dc84331 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Mon, 7 Sep 2026 16:38:06 -0500
Subject: [PATCH] winewayland: Move flush from
 wayland_client_surface_attach_image_description.

---
 dlls/winewayland.drv/vulkan.c          | 2 +-
 dlls/winewayland.drv/wayland_surface.c | 2 --
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/dlls/winewayland.drv/vulkan.c b/dlls/winewayland.drv/vulkan.c
index 4d84f33ae62..d00782c9e1b 100644
--- a/dlls/winewayland.drv/vulkan.c
+++ b/dlls/winewayland.drv/vulkan.c
@@ -175,7 +175,7 @@ static VkColorSpaceKHR wayland_vulkan_map_colorspace(VkColorSpaceKHR colorspace,
     if (!wp_image_description_v1) goto err;
 
     wayland_client_surface_attach_image_description(surface, wp_image_description_v1);
-
+    wl_display_flush(process_wayland.wl_display);
     return new;
 err:
     ERR("Failed to configure image description for client surface!\n");
diff --git a/dlls/winewayland.drv/wayland_surface.c b/dlls/winewayland.drv/wayland_surface.c
index 90c7050ac04..103ae09bf1f 100644
--- a/dlls/winewayland.drv/wayland_surface.c
+++ b/dlls/winewayland.drv/wayland_surface.c
@@ -1899,13 +1899,11 @@ void wayland_client_surface_attach_image_description(struct wayland_client_surfa
             wp_color_management_surface_v1_destroy(surface->wp_color_management_surface_v1);
             surface->wp_color_management_surface_v1 = NULL;
         }
-        wl_display_flush(process_wayland.wl_display);
         return;
     }
 
     client_surface_add_ref(&surface->client);
     wp_image_description_v1_add_listener(image_desc, &image_description_listener, surface);
-    wl_display_flush(process_wayland.wl_display);
 }
 
 /**********************************************************************
